clippy_lints/src/derivable_impls.rs

```diff
@@ -128,11 +140,15 @@ fn check_struct<'tcx>(
         _ => false,
```

Comment:
Let's return early if we won't lint anyway:

```suggestion
        _ => return,
```